### ISSUE TITLE
feat: vehicle armour and penetration overhaul (STANAG-aligned)

### ISSUE THREAD

**Issue by MBarkerUK** [lErrorl404l/AEE#126]
## Summary

Vehicle armour and penetration overhaul. The engine has TWO independent gates: `CfgVehicles >> armor` (hitpoint damage pool) and bisurf `bulletPenetrability` (does the round physically pass through). AEE should align both to real-world values. Research grounded in ACE3 source (ace_vehicle_damage), Bohemia wiki, STANAG 4569, and DeMarre.

## The Two Gates

1. **Armor** (`CfgVehicles >> armor`): a health pool. Formula: `damageResistance = (HIT^2)/Armor x (0.27/tgtRadius)^2`. Per-hitpoint armor is a multiplier (engine 0.8, turret 0.8, tracks 0.6, wheels 0.05, glass 0.5).
2. **bisurf bulletPenetrability** (from #124): does the round pass? Engine: `mm penetration = (v/1000) x caliber x bulletPenetrability`. Caliber 1 = 15 mm RHA at 1000 m/s.

Two gates: `armor` decides damage, bisurf decides passage.

## Vanilla Armor Values (verify in-game before overriding)

| Class | Armor |
|---|---|
| Infantry | 2-3 |
| Motorcycle | 10 |
| Hatchback/offroad | 20 |
| Truck | 40 |
| MRAP | 60 |
| IFV/APC | 400 |
| MBT | 800 |
| Helicopter transport | 40 |
| Attack heli/fighter | 60 |
| Armed boat | 300 |

## STANAG 4569 Protection Levels (the target table)

| Level | Threat | ~RHA |
|---|---|---|
| L1 | 7.62x51 Ball 833 m/s | ~4.7 mm |
| L2 | 7.62x39 API 695 m/s | ~8 mm |
| L3 | 7.62x51 AP 930 m/s | ~18 mm |
| L4 | 14.5x114 API 911 m/s | ~28 mm |
| L5 | 25x137 APDS-T 1258 m/s | - |
| L6 | 30 mm APFSDS | - |

Modern MBT: Leopard 2A4 590-690 mm RHAe KE, 2A6 920-940 mm, T-72B ~510 mm, T-90 w/Kontakt-5 ~690 mm.

## Penetration Models

DeMarre scaling law:
```
P = P_ref x (V/V_ref)^1.4283 x (D/D_ref)^1.0714 x (W/D^3)^0.7143 / (W_ref/D_ref^3)^0.7143
```
LOS thickness: `thickness / cos(angle from normal)`. Angle handled natively by the engine's bisurf penetration.

## Scripted Hooks (the dynamic path)

- `HitPart` EH: `["_projectile","_hitEntity","_projectileOwner","_pos","_velocity","_normal","_components","_radius","_surfaceType"]`
- `Penetrated` EH: `["_projectile","_hitObject","_surfaceType","","_spallPosASL","_newVelocity"]` — speed loss = `velocity _projectile vectorDiff _newVelocity`
- `HandleDamage` EH for full damage replacement

## What ACE3 Already Does (DO NOT duplicate)

ACE3 `ace_vehicle_damage`:
- Overrides engine damage via HandleDamage EH, queues hits, probabilistic test on `minimalHit`: `_penChance = _addedDamage/_minDamage`
- Scales AP damage by impact speed
- Classifies warheads by `warheadName`
- Spalling (frag addon)
- Material classification (`fnc_getMaterialInfo`)

Use the double-count guard pattern: return 1.0 when `ace_vehicle_damage` is active.

## Honest Scope Ranking

1. **Armor config overrides** to STANAG-aligned values — HIGH value, LOW effort, SHIP FIRST. Per-class re-alignment: IFV 400 -> ~L4/L5, MRAP 60 -> L2/L3, MBT 800 -> realistic RHAe-tuned. This fixes the "any rifle shreds an IFV" problem with a config patch.
2. **Material library** (#124) — the bisurf gate, already issued. Rank 2.
3. **Dynamic penetration model** (DeMarre + overmatch on HitPart) — medium-high value, high effort. The engine already does angle-aware bisurf penetration; only add overmatch and behind-armour effects.
4. **Physical spalling** — ACE3 ships it; only add physical spall velocity via the #107 fragment model.

## Implementation Steps

1. Ship `aee_bugfix` (or new `aee_armour`) config overrides for Armor values, gated behind a setting
2. Align per-hitpoint armor multipliers where vanilla is wrong
3. Verify vanilla values in-game: `getNumber (configFile >> "CfgVehicles" >> _class >> "armor")`
4. Wire the `Penetrated` EH (in #124's apply path) to feed #107 fragmentation for spall
5. Later: dynamic DeMarre overmatch model on HitPart, gated behind ACE3 guard

## Caveats

- `armor` is a health pool, NOT thickness — don't confuse the two
- Vanilla values need in-game verification before override
- bisurf plate values live in data_f.pbo (extract or test in-game)
- MBT front turret in-game will never match real 900 mm RHAe at game scale — tune to gameplay + plausible

## Labels
ballistics, P2-medium, effort/M, type/feat, phase/2-expansion, mobility, core

**Comment by MBarkerUK** [comment]
Delivered in 411933f+ (merged PR #211): the STANAG-aligned armour overhaul with a fully research-backed basis. Key findings: the two gates (armor pool + bisurf penetration) are INDEPENDENT (not multiplicative); the engine caliber is normalised (7.62 Ball = 18.7mm RHA at muzzle, vanilla over-penetrates); the vanilla ladder was verified locally (APC_Tracked 500/900, not the issue's 400/800); ACE3 does NOT own the armour config surface so this coexists. aee_armour re-aligns the pool ladder (MRAP->160, IFV->420/600, MBT->1100) and fnc_penetrationGate scales damage by the round's real RHA penetration vs the STANAG class. 13 tests lock it. DeMarre model + spalling remain for the follow-on (#107 fragment model).
